buildSrc/src/main/java/quilt/internal/tasks/diff/DownloadTargetMappingJarTask.java

```diff
@@ -0,0 +1,83 @@
+package quilt.internal.tasks.diff;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import quilt.internal.Constants;
+import quilt.internal.tasks.DefaultMappingsTask;
+
+public class DownloadTargetMappingJarTask extends DefaultMappingsTask {
+    public static final String TASK_NAME = "downloadTargetMappingsJar";
+
+    @OutputFile
+    private final RegularFileProperty targetMappingsFile;
+
+    @OutputFile
+    private final RegularFileProperty targetUnpickFile;
+
+    @OutputFile
+    private final RegularFileProperty targetUnpickDefinitionsFile;
+
+    @OutputFile
+    private final RegularFileProperty targetUnpickConstantsFile;
+
+    public static final String TARGET_MAPPINGS = ".gradle/targets";
+
+    public DownloadTargetMappingJarTask() {
+        super("diff");
+
+        CheckTargetVersionExistsTask checkExists = this.getTaskByType(CheckTargetVersionExistsTask.class);
+        this.onlyIf(task -> checkExists.getTargetVersion().isPresent());
+
+        this.dependsOn(CheckTargetVersionExistsTask.TASK_NAME);
+
+        targetMappingsFile = getProject().getObjects().fileProperty();
+        targetUnpickFile = getProject().getObjects().fileProperty();
+        targetUnpickDefinitionsFile = getProject().getObjects().fileProperty();
+        targetUnpickConstantsFile = getProject().getObjects().fileProperty();
+        targetMappingsFile.convention(() -> getProject().file(TARGET_MAPPINGS + "/quilt-mappings-" + checkExists.getTargetVersion().orElse(Constants.MAPPINGS_VERSION) + "/mappings/mappings.tiny"));
```

Comment:
`TARGET_MAPPINGS + "/quilt-mappings-" + checkExists.getTargetVersion().orElse(Constants.MAPPINGS_VERSION)` should be in a variable

Reply:
It cant be in a variable because the `.get` for the optional must be lazy loaded

Reply:
What if you put it in a supplier instead?